Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -5013,14 +5013,18 @@ private void PrioritizeAndSchedule(LoggingContext loggingContext, IEnumerable<No
                             // Below, we add one or more quanitites in the uint range.
                             // We use a long here to trivially avoid any overflow, and saturate to uint.MaxValue if needed as the last step.
                             long criticalPath = 0;
+                            int priorityBase = 0;
 
                             // quick check to avoid allocation of enumerator (as we are going through an interface, and where everything gets boxed!)
                             if (!graph.IsSinkNode(node))
                             {
                                 foreach (var edge in graph.GetOutgoingEdges(node))
                                 {
-                                    var otherCriticalPath = GetPipRuntimeInfo(edge.OtherNode).Priority;
-                                    criticalPath = Math.Max(criticalPath, otherCriticalPath);
+                                    var otherPriority = GetPipRuntimeInfo(edge.OtherNode).Priority;
+
+                                    // Priority consists of given priority in the specs (bits 24-31, and the critical path priority (bits 0-23)
+                                    criticalPath = Math.Max(criticalPath, otherPriority & ((1 << 24) - 1));
+                                    priorityBase = Math.Max(priorityBase, otherPriority >> 24);

```

Comment:
>24 [](start = 91, length = 2)

Can this number `24` be represented as a constant? So that changing one will entails changing the others.